saleor/dashboard/product/forms.py

```diff
@@ -138,9 +143,29 @@ class Meta:
 
     def __init__(self, *args, **kwargs):
         super(ProductImageForm, self).__init__(*args, **kwargs)
+        if self.instance.product:
+            variants = self.fields['variants']
+            variants.queryset = self.instance.product.variants.all()
+            variants.initial = self.instance.variant_images.values_list(
+                'variant', flat=True)
         if self.instance.image:
             self.fields['image'].widget = ImagePreviewWidget()
 
+    def save_variant_images(self, instance):
+        variant_images = []
+        # Clean up old mapping
+        with transaction.atomic():
```

Comment:
Since this function only does one thing, you can decorate it instead of using the context manager.